Design (#84)
chhoumann/p7: report/sections/Chapter 04 - Design/design.tex

> Based on our specification we have created a design for the application's user interface (UI). The UI had to support the use cases specified in section \ref*{sec:use_cases}, although not all features required in the use cases were designed for. The goal of the design was to include the features listed as \textit{Must have} and \textit{Should have} in the MoSCoW analysis, while including some of the features from the other categories. This was done to ensure that the most important aspects of the use cases would be satisfied in the design, while also having designed for some of the desirable, but less important features.

Reviewer: ```suggestion
Based on our specification we created a design for the application's user interface (UI). The UI had to support the use cases specified in section \ref*{sec:use_cases}, although not all features required in the use cases were included in the design. The goal of the design was to include the features listed as \textit{Must have} and \textit{Should have} in the MoSCoW analysis, but still take the other features into account. This was done to ensure that the most important aspects of the use cases would be satisfied in the design, while also having designed for some of the desirable, but less important features.
```
Past tense & some clarifications. Please do say if you don't agree.

\section{Wireframe for creating an exercise}
The wireframe in figure \ref{fig:wfProblem}, shows the page that the lecturer will see, when the create exercise button when browsing exercises. 
% create exercises
\begin{figure}[H]
	\includegraphics[scale=0.6]{createProblem.jpg}
	\centering
	\caption{wireframe for creating a problem}
	\label{fig:wfProblem}
\end{figure}

This page allows the lecturer to create an exercise with a title, test, function template and description. These are fields required to provide the information necessary for a student to solve the exercises. Additionally, test field allows a lecturer to specify the tests that student's code must pass. The function template field allows the lecturer to specify functions that the test will call, and which will be included in the editor on the solve exercise page. This satisfies the remaining requirements of the use case \textbf{Create exercises}.

> As mentioned previously not all the use cases have been designed for, because the remaining use cases and related features were deemed as nonessential for a minimal viable product. The next chapter will cover the system architecture.

Reviewer: I just noticed that the MVP section isn't actually included in our report so far. That is, `MVP.tex` isn't included. That makes this the first place we mention a minimal viable product. Not sure if that's great for the reader---they won't know what that is.
